In [2]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
import sklearn
from sklearn import model_selection, tree, metrics,svm
from sklearn.linear_model import LogisticRegression
from sklearn.model_selection import train_test_split, GridSearchCV, KFold, cross_val_score
from sklearn.metrics import mean_squared_error, mean_squared_log_error, roc_auc_score
from sklearn.preprocessing import Normalizer, StandardScaler
from sklearn.discriminant_analysis import LinearDiscriminantAnalysis
from math import sqrt
import warnings
warnings.filterwarnings("ignore")

train = pd.read_excel("Trucks.xlsx")
train

Unnamed: 0   Freq  period  parked  light  collision
0            1    712       0       1      1          0
1            2    613       1       1      1          0
2            3    192       0       1      1          1
3            4    179       1       1      1          1
4            5   2557       0       0      1          0
5            6   2373       1       0      1          0
6            7  10749       0       0      1          1
7            8   9768       1       0      1          1
8            9    634       0       1      0          0
9           10    411       1       1      0          0
10          11     95       0       1      0          1
11          12     55       1       1      0          1
12          13    325       0       0      0          0
13          14    283       1       0      0          0
14          15   1256       0       0      0          1
15          16    987       1       0      0          1
16          17    345       0       1      0          0
17          18    179       1       1      0          0
18          19     46       0       1      0          1
19          20     39       1       1      0          1
20          21    579       0       0      0          0
21          22    494       1       0      0          0
22          23   1018       0       0      0          1
23          24    885       1       0      0          1

### Question 8. What is the percentage prior probability of the occurrence of '1' in the predicted variable?

#### Option a is the answer(50)

In [3]:
X = train.drop(['Unnamed: 0', 'collision'],axis=1)
y = train['collision']

In [4]:

model = LinearDiscriminantAnalysis()

model.fit(X, y)



LinearDiscriminantAnalysis()

In [5]:
y_pred = model.predict(X)

In [6]:
metrics.classification_report(y_true=y, y_pred=y_pred)

'              precision    recall  f1-score   support\n\n           0       0.57      0.67      0.62        12\n           1       0.60      0.50      0.55        12\n\n    accuracy                           0.58        24\n   macro avg       0.59      0.58      0.58        24\nweighted avg       0.59      0.58      0.58        24\n'

### Question 9. What is the f1 score of the Linear Discriminant Analysis model with '0' as the positive class?

#### Option a is the answer(0.62)

### Question 10. Higher the weight of the variable in a standardized predictor environment, we can say that the particular variable has a higher discriminating power.

#### Option a is the answer(True)